airsonic-main/src/main/java/org/airsonic/player/service/upnp/ArtistUpnpProcessor.java

```diff
@@ -28,13 +28,15 @@
 import org.fourthline.cling.support.model.container.Container;
 import org.fourthline.cling.support.model.container.MusicArtist;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
 
 import java.util.List;
 
 /**
  * @author Allen Petersen
  * @version $Id$
  */
+@Service
 public class ArtistUpnpProcessor extends UpnpContentProcessor <Artist, Album> {
```

Comment:
Is it a @service or a @component ? Same question for all these kind of classes in upnp package.

Reply:
@service is a @component you really can use either in this case. Service i usually think of as business logic or layer that uses other services or dao/data layer. Component is the more generic. Service is a component and a controller is a component it is just the generic for spring.